src/entities/nestedJoin/getNestedJoinCalls.ts

```diff
@@ -0,0 +1,70 @@
+import { Token } from '../token';
+import { getPoolAddress } from '../../utils';
+import { NestedJoinInput, NestedJoinCall } from './types';
+import { NestedPoolState } from '../types';
+
+export const getNestedJoinCalls = (
```

Comment:
What do you think about changing the naming for this? I found it a bit confusing because its not really creating any calls. Really its getting pool data that can be used to build a call later. Specifically a query call with min out of 0.

Reply:
The way I was interpreting the result was as non-encoded calls, which would then be formatted (through getParsedCalls) and then encoded, but I agree that it's a bit confusing.
And I agree it should mention something about being query data, since it's setting minAmountsOut to 0.
With that in mind, I'll rename it to `getQueryCallData` (no need to keep NestedJoin within the name, since it's being used internally only)
Let me know if you have a better suggestion 😉

Reply:
Actually `callData` didn't seem to represent an array of calls attributes, so I decided to go with `getQueryCallsAttributes`.